app/server/lib/ConfigBackendAPI.ts

```diff
@@ -0,0 +1,67 @@
+import * as express from 'express';
+import {expressWrap} from 'app/server/lib/expressWrap';
+
+import {IGristCoreConfig} from 'app/server/lib/configCore';
+import {getGlobalConfig} from 'app/server/lib/globalConfig';
+
+import log from "app/server/lib/log";
+
+export class ConfigBackendAPI {
+  private static _doRestart = false;
+  private _config: IGristCoreConfig;
+
+  constructor() {
+      this._config = getGlobalConfig();
+  }
+
+  public addEndpoints(app: express.Express, requireMiddleware: express.RequestHandler) {
+    app.get('/api/config/:key', requireMiddleware, expressWrap((req, resp) => {
+      log.debug(`Scheduled to do a config restart is ${ConfigBackendAPI._doRestart}`);
+      if (ConfigBackendAPI._doRestart) {
+        resp.on('finish', () => {
```

Comment:
Can you explain this bit `resp.on('finish')`. I think this should be post method, as there are some sideeffects.
Or maybe we can expose dedicated endpoint for restarting?

Sorry for all this question :) I haven't seen that code before.

Reply:
I had it as a dedicated enpoint, but it worked essentially the same way.

The problem is that you want to restart the server, but you also want to respond to the frontend when it sends a request. So, you have to execute the restart code after the request has been sent.

What is a "post method"? Is it different than using an event listener on the response?

Reply:
I've reworked the logic to include a separate `api/admin/restart` endpoint instead of removing that endpoint.